lib/_stream_writable.js

```diff
@@ -496,30 +478,25 @@ function onwrite(stream, er) {
       onwriteError(stream, state, er, cb);
     }
   } else {
-    // Check if we're actually ready to finish, but don't emit yet
-    var finished = needFinish(state) || stream.destroyed;
-
-    if (!finished &&
-        !state.corked &&
-        !state.bufferProcessing &&
-        state.bufferedRequest) {
-      clearBuffer(stream, state);
-    }
-
-    if (sync) {
-      // It is a common case that the callback passed to .write() is always
-      // the same. In that case, we do not schedule a new nextTick(), but rather
-      // just increase a counter, to improve performance and avoid memory
-      // allocations.
-      if (state.afterWriteTickInfo !== null &&
-          state.afterWriteTickInfo.cb === cb) {
-        state.afterWriteTickInfo.count++;
+    clearBuffer(stream, state);
```

Comment:
This no longer checks for `stream.destroyed` before calling `clearBuffer`. Is this to be handled by emptying the buffer by the `destroy()` below?
Also, as the `destroy()` doesn't reset the `bufferedIndex` this may lead to `buffered.length - bufferedIndex` in `clearBuffer` being negative and therefore missing `if (!bufferedLength` check.

Reply:
> This no longer checks for stream.destroyed before calling clearBuffer. Is this to be handled by emptying the buffer by the destroy() below?

Correct.